src/Components/WaitingRoom.tsx

```diff
@@ -1,19 +1,31 @@
-import { useEffect } from 'react';
+import { useEffect, useState } from 'react';
 import { useGame, assignRandomActions } from '../Contexts/GameContext';
 import { pawnDBInitialState } from '../Contexts/PawnContext';
-import { Paper, Stack, Button, List, ListItem, Alert, Box } from '@mui/material';
-import { DBPlayer } from '../types';
+import { Paper, Stack, Button, List, ListItem, Alert, Box, Modal, Typography } from '@mui/material';
+import { DBPlayer, playerNumber } from '../types';
 import { setDoc, useDocData } from "../utils/useFirestore"; 
 import { allTiles } from '../Data/all-tiles-data';
 import { doc } from '../utils/useFirestore'
-import {deleteDoc} from "firebase/firestore";
+import { deleteDoc, updateDoc } from "firebase/firestore";
+import CloseIcon from '@mui/icons-material/Close';
+import { useRoomHostDocState } from '../Contexts/FirestoreContext';
 
-const WaitingRoom = ({isHost}: {isHost: boolean}) => {
+const WaitingRoom = ({currentPlayer}: {currentPlayer: any}) => {
   const { gameState, gameDispatch } = useGame();
 
+  const [open, setOpen] = useState<boolean>(false)
```

Comment:
you can add here `const { player } = usePlayerDocState()`
access "currentPlayer" within `player`